whistl/utils.py

```diff
@@ -7,6 +7,30 @@
 import numpy as np
 
 
+def count_items_in_dataloader(dataloader):
+    '''Calculate the total number of items in a dataset by iterating through a dataloader
+
+    Arguments
+    ---------
+    dataloader: pytorch DataLoader object
+        The dataloader to be iterated over
+
+    Returns
+    -------
+    item_count: int
+        The number of items in the dataset
+    '''
+    item_count = 0
+    for batch in dataloader:
+        # TODO this should be safe since there should always be at least one thing returned right?
```

Comment:
You could use a try/except block here if you're worried about empty batches. Maybe something like:
```python
try:
    X = batch[0]
    item_count += X.shape[0]
except KeyError:
    pass
```

Reply:
Good idea!